postgres/lateral_expression.go

```diff
@@ -0,0 +1,41 @@
+package postgres
+
+import (
+	"github.com/go-jet/jet/v2/internal/jet"
+)
+
+type LateralExpression interface {
+	As(alias string) LateralTable
```

Comment:
You can move `As` from this interface into bellow `LateralTable` interface and remove this one. 
Also use capital case for SQL keywords - `AS`.

Reply:
Yeah sorry my IDE changed that...

LATERAL expression recieves a SelectStatment, but to enforce the use of ALIAS it returns LateralExpression, so it nows requires to use the AS method to return a LateralTable

Reply:
Ok, it makes sense.